File: threedollar-api-user-service/src/main/java/com/depromeet/threedollar/api/user/controller/advice/WebDataBinderAdvice.java
package com.depromeet.threedollar.api.user.controller.advice;

import org.springframework.beans.propertyeditors.StringTrimmerEditor;
import org.springframework.web.bind.WebDataBinder;
import org.springframework.web.bind.annotation.InitBinder;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class WebDataBinderAdvice {

    @InitBinder
    public void initBinder(WebDataBinder binder) {
        binder.initDirectFieldAccess();
        binder.registerCustomEditor(String.class, new StringTrimmerEditor(false));
    }

}

File: threedollar-api-boss-service/src/main/kotlin/com/depromeet/threedollar/api/boss/controller/advice/WebDataBinderAdvice.kt
package com.depromeet.threedollar.api.boss.controller.advice

import org.springframework.beans.propertyeditors.StringTrimmerEditor
import org.springframework.web.bind.WebDataBinder
import org.springframework.web.bind.annotation.InitBinder
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class WebDataBinderAdvice {

    @InitBinder
    fun initBinder(binder: WebDataBinder) {
        binder.initDirectFieldAccess()
        binder.registerCustomEditor(String::class.java, StringTrimmerEditor(false))
    }

}
